src/Workspaces/Core/Portable/Utilities/NullableHelpers/NullableExtensions.cs

```diff
@@ -6,119 +6,32 @@ namespace Microsoft.CodeAnalysis
 {
     internal static partial class NullableExtensions
     {
-        public static T WithNullability<T>(this T typeSymbol, NullableAnnotation nullability) where T : class, ITypeSymbol
-        {
-            if (typeSymbol == null)
-            {
-                return null;
-            }
-
-            if (typeSymbol is TypeSymbolWithNullableAnnotation typeSymbolWithNullability)
-            {
-                // Check if the wrapper already has the same top-level nullability; in that case we don't need to re-create one.
-                if (typeSymbolWithNullability.Nullability == nullability)
-                {
-                    return typeSymbol;
-                }
-
-                // No reason to wrap a wrapper, so unwrap it
-                typeSymbol = (T)typeSymbolWithNullability.WrappedSymbol;
-            }
-
-            return typeSymbol switch
-            {
-                IArrayTypeSymbol arrayTypeSymbol => (ITypeSymbol)new ArrayTypeSymbolWithNullableAnnotation(arrayTypeSymbol, nullability),
-                IDynamicTypeSymbol dynamicTypeSymbol => (ITypeSymbol)new DynamicTypeSymbolWithNullableAnnotation(dynamicTypeSymbol, nullability),
-                INamedTypeSymbol namedTypeSymbol => (ITypeSymbol)new NamedTypeSymbolWithNullableAnnotation(namedTypeSymbol, nullability),
-                IPointerTypeSymbol pointerType => (ITypeSymbol)new PointerTypeSymbolWithNullableAnnotation(pointerType, nullability),
-                ITypeParameterSymbol typeParameterSymbol => (ITypeSymbol)new TypeParameterSymbolWithNullableAnnotation(typeParameterSymbol, nullability),
-                _ => throw ExceptionUtilities.UnexpectedValue(typeSymbol)
-            } as T;
-        }
-
         public static T WithNullability<T>(this T typeSymbol, NullableFlowState flowState) where T : class, ITypeSymbol
         {
             // TODO: call the compiler API once it's available
```

Comment:
❔ Comment still relevant?

Reply:
Yes, because that code will still go away. The compiler is still fixing up that bit and then we'll have the full cleanup.

(This PR originally deleted this too, as I was unaware that the compiler work was still forthcoming. Had to split it up but a new PR will be created once this one is merged.)